src/main/java/io/quarkus/security/PermissionChecker.java

```diff
@@ -47,8 +47,33 @@
  * }
  * }
  * </pre>
- * The permission checker method parameters are matched with the secured method parameters in exactly same fashion
- * as are constructor parameters of a custom permission. Please see {@link PermissionsAllowed#params()} for more information.
+ * The permission checker method parameters are matched with the secured method parameters exactly the same way as
+ * the constructor parameters of a custom permission are. Please see {@link PermissionsAllowed#params()} for more information.
+ * <pre>
+ * If the {@link PermissionsAllowed} annotation lists several permission names and its {@link PermissionsAllowed#inclusive} property is set to `true` then an equal number of permission checker methods must be available.
+ * Consider the following secured method:
+ * <pre>
+ * {@code
+ * &#64;PermissionsAllowed(value={"read", "write"}, inclusive=true)
+ * public String readWriteString(String a) {
+ *     ...
+ * }
+ * }
+ * </pre>
+ * For the access to the {@code readWriteString} method be granted, two permission checkers,
+ * one for the `read` permission, and another one for the `write` permission, must be available:
+ * <pre>
+ * {@code
+ * &#64;PermissionChecker("read")
+ * public boolean canRead(SecurityIdentity identity) {
+ *     ...
+ * }
+ * &#64;PermissionChecker("write")
```

Comment:
It is a special case, but I suppose you could also add an example with colon so that users understand it behaves differently (no actions).

Reply:
@michalvavrik Sure, I updated one of the permissions to have `read:all` and added a note in the very end that the match is achieved if the String names are equal